src/actions/cart-manager.ts

```diff
@@ -0,0 +1,92 @@
+import {retrieveSpecificProduct} from './DAL/product-db-actions';
+
+export const mainCart: { id: string, users: { userId: string, amount: number }[] }[] = [];
```

Comment:
why not to use Map? then you will get, edit and delete cart in O(1) and it will save you a **lot** of code

Reply:
done